scripts/generate/templates/react.template.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * Extend Shopify Checkout with a custom Post Purchase user experience
+ * This Shopify Checkout template provides two extension points:
+ *  1. Inquiry - Called first, during the payment step.
+ *  2. Render - If requested by `Inquiry`, will be rendered after
+ *     payment processing completes
+ */
+
+import React from 'react';
+import {
+  extend,
+  renderReact,
+  InputForRenderExtension,
+  Text,
+} from '@shopify/argo-checkout';
+
+/** Define any shape or type of data */
+interface Payload {
+  couldBe: 'anything' | 'everything';
+}
+
+/**
+ * Entry point for the Inquiry Extension Point.
+ *
+ * Returns a value indicating whether or not to render a PostPurchase step,
+ * and optionally allows data to be stored on the client for use in the
+ * `::Render` extension point.
+ */
+extend('Checkout::PostPurchase::Inquiry', async ({storage}) => {
+  const {render, payload} = await doInquiry();
+  if (render) {
+    await storage.update(payload);
+  }
+  return {
+    render,
+  };
+});
+
+// Simulate results of network call, etc.
+const doInquiry = async () => {
```

Comment:
Our style guide and code is all over the place on this, but I am strong advocate for function declarations. In practice it doesn’t matter here, but there’s technically no way of knowing if the `extend` call above will throw from this, since it would if that function were immediately invoked.

Reply:
agreed.  `function` gives the lowest barrier to entry and best experience for most devs without throwing a potential curveball at them.